api/src/main/java/io/druid/timeline/DataSegmentUtils.java

```diff
@@ -0,0 +1,188 @@
+/*
+ * Licensed to Metamarkets Group Inc. (Metamarkets) under one
+ * or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. Metamarkets licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.druid.timeline;
+
+import com.google.common.base.Function;
+import com.metamx.common.logger.Logger;
+import org.joda.time.DateTime;
+import org.joda.time.Interval;
+import org.joda.time.format.DateTimeFormatter;
+import org.joda.time.format.ISODateTimeFormat;
+
+import java.util.Objects;
+
+/**
+ * identifier to DataSegment.
+ */
+public class DataSegmentUtils
+{
+  private static final Logger LOGGER = new Logger(DataSegmentUtils.class);
+
+  public static Function<String, Interval> INTERVAL_EXTRACTOR(final String datasource)
+  {
+    return new Function<String, Interval>()
+    {
+      @Override
+      public Interval apply(String identifier)
+      {
+        return valueOf(datasource, identifier).getInterval();
+      }
+    };
+  }
+
+  // ignores shard spec
+  public static SegmentElements valueOf(String dataSource, String identifier)
+  {
+    SegmentElements segmentDesc = parse(dataSource, identifier);
+    if (segmentDesc == null) {
+      throw new IllegalArgumentException("Invalid identifier " + identifier);
+    }
+    return segmentDesc;
+  }
+
+  private static SegmentElements parse(String dataSource, String identifier)
+  {
+    if (!identifier.startsWith(dataSource + DataSegment.delimiter)) {
+      return null;
+    }
+    String remaining = identifier.substring(dataSource.length() + 1);
+    String[] splits = remaining.split(DataSegment.delimiter);
+    if (splits.length < 2) {
+      return null;
+    }
+
+    DateTimeFormatter formatter = ISODateTimeFormat.dateTime();
```

Comment:
Basically it's doing the same thing but I thought that would be a little faster than `new DataTime(value)`.